File: leet2428/leet2428.py
# https://leetcode.com/problems/maximum-sum-of-an-hourglass/description/

from typing import List

class Solution:
    def maxSum(self, grid: List[List[int]]) -> int:
        m = len(grid)
        n = len(grid[0])
        max_sum = -float("inf")
        for i in range(1, m - 1):
            for j in range(1, n - 1):
                cur_sum = (
                    grid[i - 1][j - 1]
                    + grid[i - 1][j]
                    + grid[i - 1][j + 1]
                    + grid[i][j]
                    + grid[i + 1][j - 1]
                    + grid[i + 1][j]
                    + grid[i + 1][j + 1]
                )
                max_sum = max(max_sum, cur_sum)
        return max_sum

File: leet2428/leet2428.cpp
// https://leetcode.com/problems/maximum-sum-of-an-hourglass/description/

#include <vector>
using namespace std;

class Solution {
public:
    int maxSum(vector<vector<int>>& grid) {
        int m = grid.size();
        int n = grid[0].size();
        int max_sum = INT_MIN;
        for(int i = 1; i < m - 1; i++){
            for(int j = 1; j < n - 1; j++){
                int current_sum = grid[i-1][j-1] + grid[i-1][j] + grid[i-1][j+1] +
                                  grid[i][j] +
                                  grid[i+1][j-1] + grid[i+1][j] + grid[i+1][j+1];
                max_sum = max(max_sum, current_sum);
            }
        }
        return max_sum;
    }
};